pkg/sdk/testint/applications_gen_integration_test.go

```diff
@@ -0,0 +1,345 @@
+	})
+
+	t.Run("create application: version and patch", func(t *testing.T) {
+		applicationPackageName, version, patch := "hello_snowflake_package_test", "V001", 0
+		createApplicationPackageHandle(t, applicationPackageName, version, patch, false)
+
+		id := sdk.NewAccountObjectIdentifier(random.StringN(4))
+		pid := sdk.NewAccountObjectIdentifier(applicationPackageName)
+		vr := sdk.NewApplicationVersionRequest().WithVersionAndPatch(sdk.NewVersionAndPatchRequest(version, &patch))
+		comment := random.StringN(4)
+		request := sdk.NewCreateApplicationRequest(id, pid).
+			WithDebugMode(sdk.Bool(true)).
+			WithComment(&comment).
+			WithVersion(vr)
+		err := client.Applications.Create(ctx, request)
+		require.NoError(t, err)
+		t.Cleanup(cleanupApplicationHandle(id))
+
+		e, err := client.Applications.ShowByID(ctx, id)
+		require.NoError(t, err)
+		require.Equal(t, id.Name(), e.Name)
+		require.Equal(t, "ACCOUNTADMIN", e.Owner)
+		require.Equal(t, comment, e.Comment)
+		require.Equal(t, "APPLICATION PACKAGE", e.SourceType)
+		require.Equal(t, applicationPackageName, e.Source)
+		require.Equal(t, version, e.Version)
+		require.Equal(t, patch, e.Patch)
+	})
+
+	t.Run("create application: version directory", func(t *testing.T) {
+		applicationPackageName, version, patch := "hello_snowflake_package_test", "V001", 0
+		stage := createApplicationPackageHandle(t, applicationPackageName, version, patch, false)
+
+		id := sdk.NewAccountObjectIdentifier(random.StringN(4))
+		pid := sdk.NewAccountObjectIdentifier(applicationPackageName)
+		vr := sdk.NewApplicationVersionRequest().WithVersionDirectory(sdk.String("@" + stage.ID().FullyQualifiedName()))
+		comment := random.StringN(4)
+		request := sdk.NewCreateApplicationRequest(id, pid).
+			WithDebugMode(sdk.Bool(true)).
+			WithComment(&comment).
+			WithVersion(vr).
+			WithTag([]sdk.TagAssociation{
+				{
+					Name:  tagTest.ID(),
+					Value: "v1",
+				},
+			})
+		err := client.Applications.Create(ctx, request)
+		require.NoError(t, err)
+		t.Cleanup(cleanupApplicationHandle(id))
+
+		e, err := client.Applications.ShowByID(ctx, id)
+		require.NoError(t, err)
+		require.Equal(t, id.Name(), e.Name)
+		require.Equal(t, "ACCOUNTADMIN", e.Owner)
+		require.Equal(t, comment, e.Comment)
+		require.Equal(t, "APPLICATION PACKAGE", e.SourceType)
+		require.Equal(t, applicationPackageName, e.Source)
+		require.Equal(t, "UNVERSIONED", e.Version)
+		require.Equal(t, patch, e.Patch)
+	})
+
+	t.Run("show application: with like", func(t *testing.T) {
+		applicationPackageName, version, patch := "hello_snowflake_package_test", "V001", 0
+		_, e := createApplicationHandle(t, applicationPackageName, version, patch, false, true, false)
+		packages, err := client.Applications.Show(ctx, sdk.NewShowApplicationRequest().WithLike(&sdk.Like{Pattern: &e.Name}))
+		require.NoError(t, err)
+		require.Equal(t, 1, len(packages))
+		require.Equal(t, *e, packages[0])
+	})
+
+	t.Run("alter application: set", func(t *testing.T) {
```

Comment:
what about `SHARE_EVENTS_WITH_PROVIDER`? This one I would like to see in the tests, because it's not supported in the CREATE.

Reply:
getting an error message: `share_events_with provider can only be set/unset if the app is created in a different account from the app package` cross account resource creation would make this a challenging test to write. i added a todo, but I don't think it adds much value for the time being.

Reply:
Ok! Please add a subtask for SNOW-933289 and add the issue number to the comment please. We will get back to it when the time comes.

Reply:
SNOW-1016268